Host paths as iSCSI LUNs

fabiand opened the pull request.

stu-gott commented on images/iscsi-demo-target-tgtd/README.md:
```diff
@@ -87,6 +87,20 @@ $ kubectl run --rm -it qemu-test --image=kubevirt/libvirtd -- \
 
+## Exporting host paths as LUNs
+
+Sometimes OS images are large or block devices., such that you do not want
```
nit: typo ".,"

stu-gott commented on images/iscsi-demo-target-tgtd/README.md:
```diff
@@ -87,6 +87,20 @@ $ kubectl run --rm -it qemu-test --image=kubevirt/libvirtd -- \
+(`/`) into the `/host` path inside the container.
+
+You can for example use `EXPORT_HOST_DEVICE=/dev/sda /home/alice/beos.img`
+to export `dev/sda` and `/home/alice/beos.img` from the host.
```
nit: typo "/dev/sda"

stu-gott commented on images/libvirtd/run-libvirtd.sh:
```diff
@@ -3,7 +3,7 @@
 
 fatal() { echo "FATAL: $@" >&2 ; exit 2 ; }
-[[ -f /host/var/run/libvirtd.pid ]] && fatal "libvirtd seems to be running on the host"
+#[[ -f /host/var/run/libvirtd.pid ]] && fatal "libvirtd seems to be running on the host"
```
Was it intentional to comment this line?

rmohr replied:
Also not clear for me if that was intentional. The rest looks good to me.

fabiand commented:
Thanks for the comments. I addressed all of them.
The libvirtd change was intentional - but I dropped it from this PR.

This change is verified, I used it in my demo preparations.

rmohr commented:
retest this please

rmohr commented on cluster/kubectl.sh:
```diff
@@ -13,11 +13,11 @@ fi
         viewer=${SPICE_VIEWER:-remote\-viewer}
         if [ "x$3" == "x--details" ]; then
-            curl http://${master_ip}:8184/apis/kubevirt.io/v1alpha1/namespaces/default/vms/$2/spice  -H"Accept:text/plain"
+            curl -s http://${master_ip}:8184/apis/kubevirt.io/v1alpha1/namespaces/default/vms/$2/spice  -H"Accept:text/plain"
```
Was that there before already? Why not showing the errors on stdout? They are valid imho.

fabiand replied:
Fixed, adding `S` which will show errors, but keep curl otherwise silent.

rmohr replied:
Ok, my comment was confusing. It also referred the .spice.err.log

rmohr commented on images/libvirtd/Dockerfile:
```diff
@@ -26,8 +26,8 @@ RUN augtool -f /augconf
 
-COPY libvirtd.sh /libvirtd.sh
-RUN chmod a+x /libvirtd.sh
+COPY run-libvirtd.sh /run-libvirtd.sh
```
This change is still unrelated

rmohr commented on images/iscsi-demo-target-tgtd/README.md:
```diff
@@ -87,6 +87,20 @@ $ kubectl run --rm -it qemu-test --image=kubevirt/libvirtd -- \
+
+Sometimes OS images are large or block devices, such that you do not want
+to make them part of the demo image. In that case you can set
+the `EXPORT_HOST_DEVICE` environment variable to directly reference a path
```
You are using EXPORT_HOST_PATHS in the manifest below.

fabiand replied:
Thanks!

fabiand commented:
Updated, @rmohr

rmohr commented on cluster/kubectl.sh:
```diff
@@ -13,11 +13,11 @@ fi
+            curl -s http://${master_ip}:8184/apis/kubevirt.io/v1alpha1/namespaces/default/vms/$2/spice > ${KUBEVIRT_PATH}cluster/.console.vv  -H"Accept:text/plain"
             echo $viewer
-            $viewer ${KUBEVIRT_PATH}cluster/.console.vv
+            $viewer ${KUBEVIRT_PATH}cluster/.console.vv 2> .spice.err.log
```
Here, that is what I don't want ... I think it is valid to see the errors from the underlying spice tool directly on stderr

rmohr commented:
@fabiand my comments before were confusing, sorry. I don't see any reason to write remote-viewer  stderr output in a hidden file.

rmohr commented on images/iscsi-demo-target-tgtd/README.md:
```diff
@@ -87,6 +87,20 @@ $ kubectl run --rm -it qemu-test --image=kubevirt/libvirtd -- \
+
+Sometimes OS images are large or block devices, such that you do not want
+to make them part of the demo image. In that case you can set
+the `EXPORT_HOST_PATH` environment variable to directly reference a path
```
EXPORT_HOST_PATHS

rmohr commented on images/iscsi-demo-target-tgtd/README.md:
```diff
@@ -87,6 +87,20 @@ $ kubectl run --rm -it qemu-test --image=kubevirt/libvirtd -- \
+For this to work you obviously also need to bind mount in the host root
+(`/`) into the `/host` path inside the container.
+
+You can for example use `EXPORT_HOST_PATH=/dev/sda /home/alice/beos.img`
```
EXPORT_HOST_PATHS
